infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/receive/LogReplicationSinkManager.java

```diff
@@ -0,0 +1,286 @@
+import org.corfudb.runtime.CorfuRuntime;
+
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.FileReader;
+import java.io.IOException;
+import java.util.List;
+import java.util.Properties;
+import java.util.UUID;
+
+import static org.corfudb.protocols.wireprotocol.logreplication.MessageType.LOG_ENTRY_MESSAGE;
+import static org.corfudb.protocols.wireprotocol.logreplication.MessageType.SNAPSHOT_MESSAGE;
+
+/**
+ * This class represents the Log Replication Manager at the destination.
+ *
+ * It is the entry point for log replication at the receiver.
+ *
+ * */
+@Slf4j
+public class LogReplicationSinkManager implements DataReceiver {
+    private static final String config_file = "/config/corfu/corfu_replication_config.properties";
+
+    public static final int DEFAULT_READER_QUEUE_SIZE = 1;
+    public static final int DEFAULT_RESENT_TIMER = 5000;
+    public static final int DEFAULT_MAX_RETRY = 5;
+
+    private int ackCycleTime = DEFAULT_RESENT_TIMER/ DEFAULT_MAX_RETRY;
+    private int ackCycleCnt = DEFAULT_READER_QUEUE_SIZE;
+    private int bufferSize = DEFAULT_READER_QUEUE_SIZE;
+
+    private CorfuRuntime runtime;
+    private SinkBufferManager sinkBufferManager;
+    private StreamsSnapshotWriter snapshotWriter;
+    private LogEntryWriter logEntryWriter;
+
+    @Getter
+    private PersistedWriterMetadata persistedWriterMetadata;
+    private RxState rxState;
+
+    @Getter
+    @Setter
+    boolean leader;
+
+    // The roleType is active or standby
+    @Getter
+    @Setter
+    private boolean active = false;
+
+    private LogReplicationConfig config;
+    private UUID snapshotRequestId = new UUID(0L, 0L);
+
+    private int rxMessageCounter = 0;
+    private long siteConfigID = 0;
+
+
+    // Count number of received messages, used for testing purposes
+    @VisibleForTesting
+    @Getter
+    private ObservableValue rxMessageCount = new ObservableValue(rxMessageCounter);
+
+    /**
+     * Constructor Sink Manager
+     *
+     * @param localCorfuEndpoint endpoint for local corfu server
+     * @param config log replication configuration
+     */
+    public LogReplicationSinkManager(String localCorfuEndpoint, LogReplicationConfig config) {
+        this.runtime =  CorfuRuntime.fromParameters(CorfuRuntime.CorfuRuntimeParameters.builder().build())
+                .parseConfigurationString(localCorfuEndpoint).connect();
+        this.rxState = RxState.LOG_ENTRY_SYNC;
+        this.config = config;
+        init();
+    }
+
+    private void init() {
+        persistedWriterMetadata = new PersistedWriterMetadata(runtime, 0, config.getSiteID(), config.getRemoteSiteID());
+        snapshotWriter = new StreamsSnapshotWriter(runtime, config, persistedWriterMetadata);
+        logEntryWriter = new LogEntryWriter(runtime, config, persistedWriterMetadata);
+        logEntryWriter.setTimestamp(persistedWriterMetadata.getLastSrcBaseSnapshotTimestamp(),
```

Comment:
Can't this setTimestamp be part of the initialization (constructor) of LogEntryWriter?